shared/js/dataDriver.js

```diff
@@ -149,6 +152,18 @@ loop.DataDriver = function() {
             console.log(`Error handling EventSource.put: ${event.data}`, ex);
           }
           break;
+
+        // Handle an error from being unable to initially connect or
+        // unexpected disconnection of an open connection.
+        case "error":
+          if (this._sse.readyState === this._sse.CLOSED) {
+            this._sse.removeEventListener("open", this);
+            this._sse.removeEventListener("put", this);
+            this._sse.removeEventListener("error", this);
+            setTimeout(() => this._connectToRoom(this._roomToken), this._retryTimeout);
```

Comment:
Eh.. given that this is an internal API, might as well shorten this to `this._connectToRoom()` and update the definition `_connectToRoom(token = this._roomToken)` and updated doc comments for the method.